### PR TITLE
Update Datamodel 20170703_2 with diff for migration and corrections

### PR DESCRIPTION
**Pull request by sjib**
corrected fk_overflow_characteristic -> fk_overflow_char to fit to fk tablename

### CODE REVIEW

**Comment by sjib** [comment]
@3nids I did check the diff files myself today - and did a few minor corrections. Please test yourself, then we cann commit.
The views I did not put in the folder - geht them from datamodel/views
Open point: Delete all 3d geometry columns. Depending on the project, they are already out.

**Comment by m-kuhn** [comment]
Thanks for the delta scripts.

File: diff/20170703/20170703.3.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+-- FEHLER:  kann Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry nicht löschen, weil andere Objekte davon abhängen
+-- DETAIL:  Sicht qgep.vw_channel hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- Sicht qgep.vw_discharge_point hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- Sicht qgep.vw_manhole hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- Sicht qgep.vw_special_structure hängt von Tabelle qgep.od_wastewater_structure Spalte detail_geometry3d_geometry ab
+-- HINT:  Verwenden Sie DROP ... CASCADE, um die abhängigen Objekte ebenfalls zu löschen.
+
+
+--od_wastewater_structure
+ALTER TABLE qgep.od_wastewater_structure DROP COLUMN detail_geometry3d_geometry CASCADE;
```
Comment (3nids) [inline diff/20170703/20170703.3.sql]: @sjib why dropping a column here???

File: diff/20170703/20170703.3.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ALTER TABLE qgep.od_wastewater_structure ADD COLUMN detail_geometry3d_geometry geometry(CurvePolygonZ, :SRID);
+UPDATE qgep.od_wastewater_structure SET detail_geometry3d_geometry = st_force3d(detail_geometry_geometry);
+ALTER TABLE qgep.od_wastewater_structure DROP COLUMN detail_geometry_geometry CASCADE;
```
Comment (3nids) [inline diff/20170703/20170703.3.sql]: I would really avoid deltete cascade in update script. Better write manually all the elements, that would avoid removing user extra in the db without him noticing.

File: diff/20170703/20170703.3.sql [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+-- add deleted views again (versions without 3d_geometry)
+-- qgep.vw_discharge_point
+-- qgep.vw_manhole
+-- qgep.vw_special_structure
+-- qgep.vw_network_node
+-- qgep.vw_reach
+
+DO language plpgsql $$
+BEGIN
+  RAISE NOTICE 'Add Views qgep.vw_discharge_point, qgep.vw_manhole, qgep.vw_special_structure, qgep.vw_network_node and qgep.vw_reach again if they were deleted (datamodel/view)';
```
Comment (3nids) [inline diff/20170703/20170703.3.sql]: why not doing it here directly?

Comment (sjib) [reply diff/20170703/20170703.3.sql]: Ok, I will add this

**Comment by sjib** [comment]
@3nids What do you think - is this better and can we merge it?

**Comment by 3nids** [comment]
from a quick look, this seems better.
but it's a bit hard to see what's going on in so many changes...

@m-kuhn do we wait for pum to be merged?

**Comment by m-kuhn** [comment]
@marioba what's the PUM for qgep schedule?

**Comment by marioba** [comment]
@m-kuhn end of December / beginning of January

**Comment by 3nids** [comment]
@sjib what do we do here?
can you update the PR with delta scripts?

**Comment by 3nids** [comment]
closing, feel free to reopen if needed

**Pull request closed without merging**